client/src/hooks/useDatasetManager.js

```diff
@@ -8,83 +6,171 @@ import { useScPCAPortal } from './useScPCAPortal'
 export const useDatasetManager = () => {
   const { myDataset, setMyDataset, datasets, setDatasets, email, setEmail } =
     useContext(DatasetManagerContext)
-  const { token, userFormat, setUserFormat } = useScPCAPortal()
+  const { token } = useScPCAPortal()
 
   /* Dataset-level */
-  const clearDataset = async (dataset) => {
-    const updatedDataset = await updateDataset({ ...dataset, data: {} })
-    return updatedDataset
-  }
+  const isMyDatasetId = (dataset) =>
+    myDataset.id !== null && dataset.id === myDataset.id
 
   const createDataset = async (dataset) => {
-    // Asserts that format is defined and save if defined
-    const newDataset = await api.datasets.create(dataset, token)
-    // On success, update myDataset
-    return newDataset
+    // TODO: Component is reponsible for generating a valid token before request
+    // Assumption: If myDataset creation initiated via UI, token should exist.
+    if (!token) {
+      console.error('A valid token is required to create a dataset')
+      throw new Error('Token required') // if necessary
+    }
+
+    // TODO: Component is reponsible for setting format before request
+    // Assumption: If myDataset creation initiated via UI, format should exist.
+    if (!dataset.format) {
+      console.error('A format is required to create a dataset.')
+      throw new Error('Format required') // if necessary
```

Comment:
Let's push these errors to a state variable that is defined in the context. Consumers can read them if they need to present them and for now we can just expose setErrors to clear them.